libs/blocks/footer/footer.js

```diff
@@ -1 +1,206 @@
+  }
+
+  init = async () => {
+    const wrapper = createTag('div', { class: 'footer-wrapper' });
+
+    const grid = this.decorateGrid();
+    if (grid) {
+      wrapper.append(grid);
+    }
+
+    const infoRow = createTag('div', { class: 'footer-info' });
+    const infoColumnLeft = createTag('div', { class: 'footer-info-column' });
+    const infoColumnRight = createTag('div', { class: 'footer-info-column' });
+
+    const region = await this.decorateRegion();
+    if (region) {
+      infoColumnLeft.append(region);
+      infoRow.classList.add('has-region');
+    }
+
+    const social = this.decorateSocial();
+    if (social) {
+      infoColumnLeft.append(social);
+      infoRow.classList.add('has-social');
+    }
+
+    const privacy = this.decoratePrivacy();
+    if (privacy) {
+      infoColumnRight.append(privacy);
+      infoRow.classList.add('has-privacy');
+    }
+
+    if (infoColumnLeft.hasChildNodes()) {
+      infoRow.append(infoColumnLeft);
+    }
+    if (infoColumnRight.hasChildNodes()) {
+      infoRow.append(infoColumnRight);
+    }
+    if (infoRow.hasChildNodes()) {
+      wrapper.append(infoRow);
+    }
+
+    this.el.append(wrapper);
+  };
+
+  decorateGrid = () => {
+    const gridBlock = this.body.querySelectorAll('div');
+    const footer = document.querySelector('footer');
+    if (!gridBlock) return null;
+    this.desktop.addEventListener('change', this.onMediaChange);
+    const navGrid = createTag('div', { class: 'footer-nav-grid' });
+    const columns = gridBlock;
+    const columnsArray = Array.from(columns);
+    const regionSelectorIndex = columnsArray.findIndex((column) => column.classList.contains('region-selector'));
+    const navCols = columnsArray.slice(0, regionSelectorIndex - 1);
+    navCols.forEach((column) => {
+      if (column) {
+        const navColumn = createTag('div', { class: 'footer-nav-column' });
+        const headings = column.querySelectorAll('h2');
+        headings.forEach((heading) => {
+          const navItem = createTag('div', { class: 'footer-nav-item' });
+          const titleId = heading.textContent.trim().toLowerCase().replace(/ /g, '-');
+          const title = createTag('a', {
+            class: 'footer-nav-item-title',
+            role: 'button',
+            'aria-expanded': this.desktop.matches,
+            'aria-controls': `${titleId}-menu`,
+          });
+          title.textContent = heading.textContent;
+          navItem.append(title);
+          const linksContainer = heading.nextElementSibling;
+          linksContainer.classList = 'footer-nav-item-links';
+          linksContainer.id = `${titleId}-menu`;
+          const links = linksContainer.querySelectorAll('li');
+          links.forEach((link) => {
+            link.classList.add('footer-nav-item-link');
+          });
+          navItem.append(linksContainer);
+          navColumn.append(navItem);
+        });
+        navGrid.append(navColumn);
+      }
+    });
+    if (navCols.length < 1) {
```

Comment:
why `< 1` instead of `=== 0`?

There's also `if (!navCols.length)` but some people don't like it

Reply:
I like !navCols.length thanks!